cpp/CMakeLists.txt

```diff
@@ -198,6 +210,7 @@ target_compile_definitions(cuspatial PUBLIC "SPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_${
 
 # Specify the target module library dependencies
 target_link_libraries(cuspatial PUBLIC cudf::cudf)
+target_link_libraries(cuspatial PRIVATE nvtx3-cpp)
```

Comment:
This is a `PUBLIC` dependency not a `PRIVATE` one. The nvtx3 headers are included by the cuspatial public headers and therefore consumers of cuspatial will also need nvtx3.

This will resolve the test compilation failures.

Reply:
We expected to need PUBLIC here and tried it but it caused another issue. @isVoid Can you provide the logs?

Reply:
The nvtx3 project doesn't have have install / export rules for the targets they create it looks like. So you would get errors when trying to export the cuspatial targets as they now depend on targets that have no way to be exported.

What we should do is specify `DOWNLOAD_ONLY` when getting nvtx3 and directly including `nvtxImportedTargets.cmake` ( https://github.com/NVIDIA/NVTX/blob/v3.1.0-c-cpp/CMakeLists.txt#L33 ). 

We will need to also setup install rules for this file and the nvtx3 headers, plus a custom `FINAL_CODE_BLOCK` I expect for the `rapids_export`.

I expect that this is why previously cuspatial copied nvtx3 directly, as it dramatically simplified the build-system.

Reply:
> Should we look into contributing to nvtx3 to provide install / export rules?

Yes but don't hold your breath for it to be put in a release anytime soon.  It took them about 2 years to release the last merged feature I was waiting on.